protostar-rust/src/lib.rs

```diff
@@ -134,6 +184,66 @@ fn run_tests(tests: TestsFromFile, tests_stats: &mut TestsStats) -> Result<()> {
     Ok(())
 }
 
+fn strip_path_from_test_names(test_configs: Vec<TestConfig>) -> Result<Vec<TestConfig>> {
+    let mut result = vec![];
+    for test_config in test_configs {
+        let name: String = test_config
+            .name
+            .clone()
+            .rsplit('/')
+            .next()
+            .with_context(|| format!("Failed to get test name from = {}", test_config.name))?
+            .into();
+        result.push(TestConfig {
+            name,
+            available_gas: test_config.available_gas,
+        });
+    }
+    Ok(result)
+}
+
+fn filter_tests_by_name(
+    test_name_filter: &str,
+    exact_match: bool,
+    test_configs: Vec<TestConfig>,
+) -> Result<Vec<TestConfig>> {
+    let mut result = vec![];
+    for test in test_configs {
+        if exact_match {
+            if test.name == test_name_filter {
+                result.push(test);
+            }
+        } else if test_name_contains(test_name_filter, &test)? {
+            result.push(test);
+        }
```

Comment:
Feel like a few filters (as proposed offline) would be better here, but won't insist. Nested ifs are a bit messy